source/photo_gen_matom.c

```diff
@@ -161,6 +162,20 @@ get_matom_f (mode)
     commbuffer = (char *) malloc (size_of_commbuffer * sizeof (char));
 #endif
 
+    /* if we are using the accelerated macro-atom scheme then we want to allocate an array 
+       for the macro-atom probabilities and various other quantities */
+#if (ACCELERATED_MACRO == TRUE)
+    PlasmaPtr xplasma;
+    int nrows = nlevels_macro + 1;
+    double **matom_matrix = (double **) calloc (sizeof (double *), nrows);
```

Comment:
I think there would be a memory leak here, since I can't see the pointer being free'd at the end of the function. It's probably not a problem, since we only calculate the emissivities once, so would only get calloc'd once. I think the same goes for the variable commbuffer too. Soz for the pedantic comment!

Reply:
Good spot - pedantry is needed on this stuff :) I'll look at it.

Reply:
do you know if you have to free each row of the matrix, or does free (matom_matrix) do it all?

Reply:
wondering whether all that is needed is:

```C
#ifdef MPI_ON 
  free (commbuffer);
#endif 
#if (ACCELERATED_MACRO == TRUE)
  free (matom_matrix);
#endif 
```

Reply:
I was just looking into that from my old lectue notes! I think the correct thing to do is,

```c
for(i = 0; i < nrows ++i){
  free(matom_matrix[i]);
}

free(matom_matrix);
``` 

Since if you do `free(matom_matrix)` it would only free whatever that points to, which is the array of pointers. I think when those are free'd, you can't get them back to free the memory which belong to them, so you have to free each pointer/element in matom_matrix and then you can free matom_matrix.

Reply:
I think there's the same thing in calc_matom_matrix for R_matrix and Q_matrix too.

Reply:
I've added a commit doing what you suggested Ed, could you maybe just check it looks right?

Reply:
Yes! I can't see any problems with your changes and it passed the CI test!